salvo/src/lib/builder/envoy_builder.py

```diff
@@ -93,18 +93,59 @@ def build_envoy_binary_from_source(self) -> str:
 
     return os.path.join(self._build_dir, constants.ENVOY_BINARY_TARGET_OUTPUT_PATH)
 
+  def build_su_exec(self) -> None:
+    """Run bazel build to generate the su-exec binary needed in Envoy docker images."""
+    cmd_params = cmd_exec.CommandParameters(cwd=self._build_dir)
+    cmd = "bazel build {bazel_options}".format(
+        bazel_options=self._generate_bazel_options(
+            proto_source.SourceRepository.SRCID_ENVOY
+        )
+    )
+    if not cmd.endswith(" "):
+      cmd += " "
+
+    cmd += "external:su-exec"
+    cmd_exec.run_check_command(cmd, cmd_params)
+
+  def stage_su_exec(self) -> None:
+    """Copy the su-exec binary used in the Envoy docker image
+
+    Returns:
+      None
+    """
+    dir_mode = 0o755
+    # The su-exec packaging in the Dockerfile does not use the
+    # TARGETPLATFORM suffix. The destination directory is the
+    # build_release string as used below
+    dest_path = os.path.join(self._build_dir, 'build_release')
+    if not os.path.exists(dest_path):
+      os.mkdir(dest_path, dir_mode)
+
+    cmd = "cp -fv "
+    cmd += "bazel-bin/external/com_github_ncopa_suexec/su-exec "
```

Comment:
Not sure it really matters here .. but it is possible to query the location of the bazel-bin directory via

```bash
bazel info bazel-bin
```

Reply:
The `bazel-bin` symlink is in the immediate directory where the docker image construction is happening.  I don't think we'd gain much by using the link target that's returned by `bazel info bazel-bin`.

Good to know that this is available and will keep it in mind.   Thanks!